internal/common/conversion/error_framework.go

```diff
@@ -0,0 +1,32 @@
+package conversion
+
+import (
+	"encoding/json"
+
+	"github.com/hashicorp/terraform-plugin-framework/diag"
+)
+
+type ErrBody interface {
+	Body() []byte
+}
+
+func AddJSONErrDiagnostics(msgPrefix string, err error, diags *diag.Diagnostics) {
```

Comment:
By default, the error details weren't shown in the Terraform error.
This approach marshalls the JSON response and includes it in the diagnostic.
An alternative would be to look for specific fields, but seems to be too error prone